tracer/src/Datadog.Trace/Telemetry/TelemetryFactory.cs

```diff
@@ -65,70 +65,92 @@ public ITelemetryController CreateTelemetryController(ImmutableTracerSettings tr
+                              ? new DependencyTelemetryCollector()
+                              : NullDependencyTelemetryCollector.Instance);
+
+                // we assume we never flip between v1 and v2
+                if (!settings.V2Enabled)
                 {
-                    var telemetryTransports = TelemetryTransportFactory.Create(settings, tracerSettings.ExporterInternal);
-
-                    if (!telemetryTransports.HasTransports)
-                    {
-                        log.Debug("Telemetry collection disabled: no available transports");
-                        return NullTelemetryController.Instance;
-                    }
-
-                    LazyInitializer.EnsureInitialized(
-                        ref _dependencies,
-                        () => settings.DependencyCollectionEnabled
-                                  ? new DependencyTelemetryCollector()
-                                  : NullDependencyTelemetryCollector.Instance);
-
-                    // we assume we never flip between v1 and v2
-                    if (!settings.V2Enabled)
-                    {
-                        // if we're not using V2, we don't need the config collector
-                        var oldConfig = Interlocked.Exchange(ref _configurationV2, NullConfigurationTelemetry.Instance);
-                        if (oldConfig is ConfigurationTelemetry config)
-                        {
-                            config.Clear();
-                        }
-                    }
-
-                    // if this changes, we will "lose" startup metrics, but unlikely to happen
-                    if (!settings.MetricsEnabled)
-                    {
-                        // if we're not using metrics, we don't need the metrics collector
-                        log.Debug("Telemetry metrics collection disabled");
-                        var oldMetrics = Interlocked.Exchange(ref _metrics, NullMetricsTelemetryCollector.Instance);
-                        if (oldMetrics is MetricsTelemetryCollector metrics)
-                        {
-                            // "clears" all the data stored so far
-                            metrics.Clear();
-                        }
-                    }
-
-                    // Making assumptions that we never switch from v1 to v2
-                    // so we don't need to "clean up" the collectors.
-                    if (settings.V2Enabled)
-                    {
-                        log.Debug("Creating telemetry controller v2");
-                        return CreateV2Controller(telemetryTransports, settings, discoveryService);
-                    }
-                    else
-                    {
-                        log.Debug("Creating telemetry controller v1");
-                        return CreateV1Controller(telemetryTransports, settings);
-                    }
+                    DisableConfigCollector();
```

Comment:
Could this be done line 117 to avoid having to if testing `settings.V2Enabled`

Reply:
Yeah, I just did a direct translation of the old code, but that makes sense 👍